core/src/main/scala/chisel3/internal/firrtl/IR.scala

```diff
@@ -219,7 +219,7 @@ private[chisel3] object ir {
       // NOTE: mod eq ctx.id only occurs in Target and Named-related APIs
       if (mod eq ctx.id) localName else name
   }
-  case class Slot(imm: Node, name: String) extends Arg {
+  case class Slot(imm: Arg, name: String) extends Arg {
```

Comment:
Good eye, so recall (or learn very quickly) that `Arg` is the Chisel IR Expression. `Node` is the `Arg` that wraps a `HasId`. In SRAM, we manually build some nested Args to emit the connections to the mask fields of the ports because representing those as actual Chisel types is really hard. We needed to build some Slots that didn't have real Chisel Data backing them up (thus no `HasIds`). We figured it was pretty weird that `Slot` _required_ a `HasId` (whereas other `Args` like `Index` don't), so this change makes more sense anyway and let us do what we needed to do to emit the connections for the mask.

Reply:
Follow up question: https://github.com/chipsalliance/chisel/pull/3955/files#diff-f01d2f59c3998b1bb4f17da0c509aa21205a6682b1cd42e274dca0fa92ebf33dR477 is preventing this alternative id-less `Slot` from showing up when doing a select query. This seems to make sense with your explanation.

Reply:
Yeah that is the only place where this change had any effect. @debs-sifive maybe we should write a test on what happens if you do a Select query for connections on a Module with an SRAM. These connections to the real ports of the SRAM are iffy and that Select may let users get at them...

Reply:
I played around with this a bit, and we can indeed see some of the memory ports, but it's all output stuff:
```
List(PredicatedConnect(List(),MemWrapper.foo.readPorts[0].data: Wire[UInt<8>],MemWrapper.foo_sram.R0.data: [UInt<8>],false), PredicatedConnect(List(),MemWrapper.foo.readwritePorts[0].readData: Wire[UInt<8>],MemWrapper.foo_sram.RW0.rdata: [UInt<8>],false))
```

So it looks like we can't drive anything on the actual memory with Select. If we want Select functionality with SRAM, I say we do it in a separate PR :)